### PR TITLE
IO updates: WvOut 24/32/64-bit support; binary mode IO support for signed and unsigned ints

### PR DESCRIPTION
**Pull request by gewang**
PRC-inspired fixes and updates to IO; targeting 1.5.0.1 release!

WvOut/WvOut2 can now output 24-bit int, 32-bit int, 32-bit float, and 64-bit float WAV, SND, AIFF audio files
note: the STK write thread message queue capacity was increased significantly from 32 to 1024, which should support many more concurrently WvOut/WvOut2 writes

Significant update to BINARY mode IO including FileIO and SerialIO; IO.readInt( flags ); IO.INT8/16/32 and (new) IO.UINT8/16/32 now return unsigned values; IO.SINT8/16/32/64 can be specified for signed int; added IO.readFloat( flags ) which can differentiate between reading (binary) 32-bit vs 64-bit floats

a lot of changes throughout, concentrated in IO and STK; I currently can't test the SerialIO changes (@spencersalazar please take a look?) cc @nshaheed

### CODE REVIEW

File: examples/basic/rec-auto-stereo.ck [hunk 1]
```diff
@@ -14,6 +14,8 @@ dac => WvOut2 w => blackhole;
 
 // this is the output file name
 "special:auto" => w.wavFilename;
```
Comment (spencersalazar) [inline examples/basic/rec-auto-stereo.ck:16]: would be sick (in another PR / round of feature updates) to be able to replace "special:auto" with a ChucK-native solution (basically datetime -> string support I think). Datetime is a massive yak shave but I think languages like Python got it pretty well down!

File: src/core/util_buffers.h [hunk 2]
```diff
@@ -203,38 +204,39 @@ template<typename T>
...
     // returns number of elements successfully put
     size_t put(T element)
     {
-        if((m_write + 1)%m_numElements == m_read)
+        // check for overflow
+        if( (m_write + 1)%m_numElements == m_read )
         {
-            // no space
+            // no space | 1.5.0.1 (ge) added log message
+            EM_log( CK_LOG_WARNING, "[circular-buffer]: buffer full, dropping items!" );
```
Comment (spencersalazar) [inline src/core/util_buffers.h:234]: might be nicer to report the error back up to the system that uses it and allow that to print the error as needed -- as-is, even if you saw this error you wouldnt necessarily know it was associated with WvOut or something else

**Review by spencersalazar — APPROVED** [review]
makes sense to me! I don't have a setup to test SerialIO, is there someone who can test that? Maybe someone in 250a or who uses Arduino? I can see about dusting off my hardware setup. Would also recommend building/testing on Windows as that can have some surprises with bit/bytepacking.

File: examples/basic/rec-auto-stereo.ck (hunk 1 above)
Comment (gewang) [reply examples/basic/rec-auto-stereo.ck:16]: a good yak to aspire to!

File: src/core/util_buffers.h (hunk 2 above)
Comment (gewang) [reply src/core/util_buffers.h:234]: agree; this is a stopgap; outside warning reporting would certainly be more informative. have opened issue #327

**Pull request merged**